hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -624,10 +694,10 @@ def mean(expr) -> Float64Expression:
     :class:`.Expression` of type :py:data:`.tfloat64`
         Mean value of records of `expr`.
     """
-    return sum(expr)/count(filter(lambda x: hl.is_defined(x), expr))
+    return sum(expr)/count_where(hl.is_defined(expr))
```

Comment:
You have to check if the count is 0. Otherwise, you can get `NaN` as the answer if all values are missing.

Reply:
isn't that OK?

Reply:
hmm... is that not the correct result if all the values are missing?

Reply:
I guess that is the current behavior. I'm not sure why in my group_by tests I was expecting to get None instead of nan.

```
In [3]: t.aggregate(hl.struct(foo = hl.agg.mean(hl.agg.filter(t.idx < 0, t.idx))))
Out[3]: Struct(foo=nan)
```